### PR TITLE
Mdbenjam fixing unassigned but completed

### PR DESCRIPTION
**Pull request by mdbenjam**
Currently we can hit a bug where we try to establish a claim and rails crashes. This happens when a task is completed but not assigned. When you click the next claim button, we query for unassigned tasks, assign it to ourselves, then try to find it by looking for tasks still needed to complete. We now also make sure that we query for non-completed tasks before assigning.

Also added some error checks to avoid moving into invalid task states. Some more thinking needs to be done here to better reason about task states.

Also removed creating bad data in the seeds file. All taks should be unassigned, unstarted, and uncompleted.

**Test Plan**
- [ ] Run test suite

### CODE REVIEW

File: app/models/task.rb [hunk 1]
```diff
@@ -69,8 +69,11 @@ def before_assign
   end
 
   def assign!(user)
+    print user
```
Comment (joofsh) [inline app/models/task.rb]: print!

File: db/seeds.rb [hunk 2]
```diff
@@ -44,21 +44,6 @@ def create_tasks(number)
       establish_claim = EstablishClaim.create(
         appeal: @appeals[i % num_appeals]
         )
-      if i % 4 > 0
```
Comment (joofsh) [inline db/seeds.rb]: So no tasks will be assigned/completed now in the seed data? Is there any way we could keep this functionality? I really loved the current seeds values for dev'ing locally

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -43,7 +43,7 @@ def current_user_historical_tasks
   helper_method :current_user_historical_tasks
 
   def next_unassigned_task
-    @next_unassigned_task ||= scoped_tasks.unassigned.first
+    @next_unassigned_task ||= scoped_tasks.unassigned.to_complete.first
```
Comment (joofsh) [inline app/controllers/tasks_controller.rb]: 😍

File: db/seeds.rb (hunk 2 above)
Comment (mdbenjam) [inline db/seeds.rb]: So the underlying problem here is that, we've added code in assign to disallow one user having many tasks.

Comment (joofsh) [inline db/seeds.rb]: For the Manager task page, you can immediately load it and see the X tasks that are in the "current" table, and Y tasks that are in the "completed" table. 

I don't think we need lots of tasks in either of those states, but would it be possible to have at least 1 that's in each?

**Pull request merged**